### PR TITLE
Add test of dictionary, mixed and embedded objects

### LINKED ISSUE #3726: Extend Dictionary Support for Embedded Objects

### PR DESCRIPTION
**Pull request by kneth**
## What, How & Why?

Only typed dictionaries allow embedded objects as values. Object store and object accessors are doing the heavy-lifting, and only a test is required.

This closes #3726 

## ☑️ ToDos

* ~[ ] 📝 Changelog entry~
* ~[ ] 📝 `Compatibility` label is updated or copied from previous entry~
* [x] 🚦 Tests
* ~[ ] 📝 Public documentation PR created or is not necessary~
* ~[ ] 💥 `Breaking` label has been applied or is not necessary~

*If this PR adds or changes public API's:*
* ~[ ] typescript definitions file is updated~
* ~[ ] jsdoc files updated~
* ~[ ] Chrome debug API is updated if API is available on React Native~

### CODE REVIEW

File: tests/js/dictionary-tests.js [hunk 1]
```diff
@@ -655,6 +655,60 @@ module.exports = {
...
+                dict1: {children1: { num: 2 }, children2: { num: 3 }},
+                dict2: {children1: { num: 4 }, children2: { num: 5 }}
+            });
+        });
+
+        realm.write(() => {
+            const error = new Error("Only Realm instances are supported.");
+            TestCase.assertThrows(() => {
+                realm.create(DictMixedSchema.name, {
+                    dict1: {children1: { num: 2 }, children2: { num: 3 }},
```
Comment (kraenhansen) [inline tests/js/dictionary-tests.js:698]: Seems to me like you're not asserting anything about this object, is that an oversight?

Comment (kneth) [reply tests/js/dictionary-tests.js:698]: I am checking if an exception is thrown as an embedded object cannot be used as value for mixed.

File: tests/js/dictionary-tests.js [hunk 2]
```diff
@@ -655,6 +655,60 @@ module.exports = {
         realm.write(() => realm.create(DictSchema.name, { a: { x: null } }));
         let data = realm.objects(DictSchema.name)[0].a;
         TestCase.assertEqual(data.x, null, "Should be an equals to mutable.x = null");
+        realm.close();
+    },
+
+    testDictionaryEmbeddedObject() {
+        const Child = {
```
Comment (fronck) [inline tests/js/dictionary-tests.js:662]:
```suggestion
        const ChildSchema = {
```

**Review by fronck — APPROVED** [review]
LGTM

**Review by takameyer — APPROVED** [review]
LGTM

**Pull request merged**